ownCloud/Library/CustomFilelistTableViewController.swift

```diff
@@ -0,0 +1,381 @@
+//
+//  CustomFilelistTableViewController.swift
+//  ownCloud
+//
+//  Created by Matthias Hühne on 13.05.19.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+	/*
+	* Copyright (C) 2019, ownCloud GmbH.
+	*
+	* This code is covered by the GNU Public License Version 3.
+	*
+	* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+	* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+	*
+	*/
+
+	import UIKit
+	import ownCloudSDK
+
+	class CustomFilelistTableViewController: UITableViewController, Themeable {
+
+		weak var core : OCCore?
+		var query : OCQuery
+
+		var lastTappedItemLocalID : String?
+		var items : [OCItem] = []
+		var progressSummarizer : ProgressSummarizer?
+		private var _actionProgressHandler : ActionProgressHandler?
+		var queryRefreshControl: UIRefreshControl?
+		var queryRefreshRateLimiter : OCRateLimiter = OCRateLimiter(minimumTime: 0.2)
+		var messageView : MessageView?
+		var refreshActionHandler: (() -> Void)?
+
+		// MARK: - Search
+		var searchController: UISearchController?
+
+		// MARK: - Sorting
+		private var sortBar: SortBar?
+		private var sortMethod: SortMethod {
+
+			set {
+				UserDefaults.standard.setValue(newValue.rawValue, forKey: "sort-method")
+			}
+
+			get {
+				let sort = SortMethod(rawValue: UserDefaults.standard.integer(forKey: "sort-method")) ?? SortMethod.alphabeticallyAscendant
+				return sort
+			}
+		}
+
+		func makeActionProgressHandler() -> ActionProgressHandler {
```

Comment:
`CustomFilelistTableViewController`, `SharesFilelistTableViewController` and `ClientQueryViewController` share quite a bit of code.

Would be great to either them `ClientQueryViewController` subclasses or move the shared code to a shared superclass, so any future additions or fixes in the shared code benefits all three at once.